src/Database/Attach/Resizer.php

```diff
@@ -275,36 +256,47 @@ public function resize($newWidth, $newHeight, $options = [])
         /*
          * Sanitize input
          */
-        $newWidth = (int) $newWidth;
-        $newHeight = (int) $newHeight;
+        $newWidth = (int)$newWidth;
+        $newHeight = (int)$newHeight;
 
         if (!$newWidth && !$newHeight) {
             $newWidth = $this->width;
             $newHeight = $this->height;
-        }
-        elseif (!$newWidth) {
+        } elseif (!$newWidth) {
             $newWidth = $this->getSizeByFixedHeight($newHeight);
-        }
-        elseif (!$newHeight) {
+        } elseif (!$newHeight) {
             $newHeight = $this->getSizeByFixedWidth($newWidth);
         }
 
         // Get optimal width and height - based on supplied mode.
         list($optimalWidth, $optimalHeight) = $this->getDimensions($newWidth, $newHeight);
 
+
         // Resample - create image canvas of x, y size
-        $imageResized = imagecreatetruecolor($optimalWidth, $optimalHeight);
+        if ($this->getOption('mode') == 'contain') {
+            $imageResized = imagecreatetruecolor($newWidth, $newHeight);
+        } else {
+            $imageResized = imagecreatetruecolor($optimalWidth, $optimalHeight);
+        }
 
         // Retain transparency for PNG and GIF files
-        imagecolortransparent($imageResized, imagecolorallocatealpha($imageResized, 0, 0, 0, 127));
+        $transparency = imagecolorallocatealpha($imageResized, 0, 0, 0, 127);
+
+        // fill empty image with transparent color, only required for contain mode
+        imagefill($imageResized, 0, 0, $transparency);
```

Comment:
If it's only required for the contain mode shouldn't be wrapped in an if statement to limit it to just that mode?